### ISSUE TITLE
Arrow.append to non-existent file

### ISSUE THREAD

**Issue by bkamins** [apache/arrow-julia#354]
Maybe we could consider allowing creation of a new Arrow file with `Arrow.append`? Currently it fails, so one needs to write:
```
for i in 1:10
    if isfile("out.arrow")
        Arrow.append("out.arrow", DataFrame(i=i))
    else
        Arrow.write("out.arrow", DataFrame(i=i))
    end
end
```
which could be just:
```
for i in 1:10
    Arrow.append("out.arrow", DataFrame(i=i))
end
```
But maybe there is a reason for the current design?

**Comment by quinnj** [comment]
Yeah, I think this shouldn't be too hard to support; cc @tanmaykm and @baumgold in case they have suggestions/comments about whether it would work or not.

**Comment by tanmaykm** [comment]
Yes, should be possible to support.
I can attempt a PR in a bit.

**Comment by CarloLucibello** [comment]
This should be closed as fixed in #358
